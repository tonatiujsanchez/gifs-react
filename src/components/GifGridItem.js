

import PropTypes from 'prop-types'


const GifGridItem = ({ title, img }) => {


  return (
    <div className="card animate__animated animate__fadeIn">
        <img src={ img } alt={ title } className="animate__animated animate__fadeIn" />
        <p className="animate__animated animate__fadeIn">{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired
}


export default GifGridItem