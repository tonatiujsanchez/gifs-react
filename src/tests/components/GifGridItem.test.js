

import GifGridItem from '../../components/GifGridItem'
import { shallow } from 'enzyme'



describe('Pruebas en el componente GifGridItem.js', () => { 


    const title = 'Titulo del gif'
    const img = 'http://localhost:3000/assets/img/gif.jpg'
    let wrapper = shallow( <GifGridItem title={ title } img={ img } /> )



    test('Debe mostrar <GifGridItem /> correctamente ', () => { 

        expect( wrapper ).toMatchSnapshot()    

     })



     test('Debe de tener un parrafo con el title', () => { 

        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( title )

      })



      test('Debe de tener la imagen igual al url y alt de los props', () => { 

        const image = wrapper.find('img')

        const src = image.props().src
        const alt = image.props().alt

        expect( src ).toBe( img )
        expect( alt ).toBe( title )

       })



       test('El componente debe de tener la clase .card', () => { 

            const divCard = wrapper.find('div')
            const classes = divCard.prop('className')
            
            expect( classes.includes('card') ).toBe( true )            

        })


 })


