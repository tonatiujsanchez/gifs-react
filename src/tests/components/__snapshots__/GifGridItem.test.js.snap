// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem.js Debe mostrar <GifGridItem /> correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo del gif"
    className="animate__animated animate__fadeIn"
    src="http://localhost:3000/assets/img/gif.jpg"
  />
  <p
    className="animate__animated animate__fadeIn"
  >
    Titulo del gif
  </p>
</div>
`;
